core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -394,6 +430,10 @@ private LoadTableResponse loadInternal(
 
   @Override
   public Table loadTable(SessionContext context, TableIdentifier identifier) {
+    if (!endpoints.contains(ResourcePaths.V1_LOAD_TABLE)) {
+      throw new NoSuchTableException("Unable to load table %s.%s", name(), identifier);
```

Comment:
nit: mention that it's because the server does not support the load endpoint (otherwise the message is too generic, hard to trouble-shoot).

Reply:
I had that initially included but removed it after https://github.com/apache/iceberg/pull/10929#discussion_r1715632323